client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/test/checklist-item.tsx

```diff
@@ -0,0 +1,69 @@
+/**
+ * @jest-environment jsdom
+ */
+import { render, screen } from '@testing-library/react';
+import React from 'react';
+import ChecklistItem from '../checklist-item';
+import { Task } from '../types';
+
+function getTask( taskData = {} ) {
+	const task: Task = {
+		id: 'foo_task',
+		isCompleted: false,
+		actionUrl: '#',
+		taskType: 'blog',
+		displayBadge: false,
+		title: 'Foo Task',
+	};
+
+	return { ...task, ...taskData };
+}
+
+describe( 'ChecklistItem', () => {
+	describe( 'when the task requires a badge', () => {
+		it( 'displays a badge', () => {
+			const badgeText = 'Badge Text';
+			render( <ChecklistItem task={ getTask( { displayBadge: true, badgeText } ) } /> );
+			expect( screen.getByText( badgeText ) ).toBeTruthy();
+		} );
+	} );
+
+	describe( 'when the task is complete', () => {
+		it( 'shows the task complete icon', () => {
+			render( <ChecklistItem task={ getTask( { isCompleted: true } ) } /> );
+			const taskCompleteIcon = screen.queryByLabelText( 'Task complete' );
+			expect( taskCompleteIcon ).toBeTruthy();
+		} );
+		it( 'hides the task enabled icon', () => {
+			render( <ChecklistItem task={ getTask( { isCompleted: true } ) } /> );
+			const taskEnabledIcon = screen.queryByLabelText( 'Task enabled' );
+			expect( taskEnabledIcon ).toBeFalsy();
+		} );
+		it( 'disables the task', () => {
+			render( <ChecklistItem task={ getTask( { isCompleted: true } ) } /> );
+			const taskButton = screen.queryByRole( 'link' );
+			expect( taskButton ).toHaveAttribute( 'disabled' );
+		} );
+	} );
```

Comment:
I think some cases we are missing here are when the task is complete but is set to remain enabled for revisiting.